Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderVariantListBuilder.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/base.h>
+#include <AssetBuilderSDK/AssetBuilderBusses.h>
+#include <AssetBuilderSDK/AssetBuilderSDK.h>
+
+#include <Atom/RHI.Reflect/Base.h>
+#include <Atom/RPI.Edit/Shader/ShaderVariantListSourceData.h>
+
+#include "ShaderBuilderUtility.h"
+
+namespace AZ
+{
+    namespace ShaderBuilder
+    {
+        struct AzslData; 
+
+        class ShaderVariantListBuilder
+            : public AssetBuilderSDK::AssetBuilderCommandBus::Handler
+        {
+        public:
+            AZ_TYPE_INFO_WITH_NAME_DECL(ShaderVariantListBuilder);
+
+            // TODO. Because we are merging this code in phases, we start with the "shadervariantlist2"
+            // extension which is the same "shadervariantlist" file format the we all know and love.
+            // Later we'll do another PR that actually referenced the "shadervariantlist" extension.
+            static constexpr char Extension[] = "shadervariantlist2";
+
+            static constexpr char JobKey[] = "HashedShaderVariantList";
+
+            // Setting this to false will always rebuild all variants when a .shadervariantlist changes.
+            // Enabled by default.
+            static constexpr char EnableHashCompareRegistryKey[] = "/O3DE/Atom/Shaders/ShaderVariantListBuilder/EnableHashCompare";
+            static constexpr bool EnableHashCompareRegistryDefaultValue = true;
+
+            // If a .shadervariantlist changes several times within this time period,
+            // the IsNew status of each previous variant in .shadervariantlist will be preserved.
+            static constexpr char SuddenChangeInMinutesRegistryKey[] = "/O3DE/Atom/Shaders/ShaderVariantListBuilder/SuddenChangeInMinutes";
+            static constexpr unsigned SuddenChangeInMinutesRegistryDefaultValue = 5;
```

Comment:
My initial analysis, looking forward into the future was a case of a `shadervariantlist` with, let's say 10K variants.
In a **conservative scenario** the ShaderVariantListBuilder would split the original list into 10K individual intermediate assets.
Subsequently each one of those assets would become an individual asset, and if it didn't change in content then the AssetProcessor wouldn't start that job anymore. All good and solid, except for the 10K File Open/Save/Close operations.

To overcome the problem There's the idea of hashing variants, doing a single file I/O operation, BUT now we have to implement a little bit of fuzzy logic (hence the sudden change logic, etc).

Long story short based from @santorac's and your feedback, it seems the clean way moving forward for now is to simply split the assets per the  **conservative scenario** mentioned above and if there are bottle-necks we can always revisit this idea of hashes and having a single `hashedvariantlist` with hashes per variant to avoid redundant asset compilation.